LLMNeedleHaystackTester.py

```diff
@@ -166,28 +146,13 @@ async def run_test(self):
         # Wait for all tasks to complete
         await asyncio.gather(*tasks)
 
-    def generate_prompt(self, context):
-        if self.model_provider == "Anthropic":
-            with open('Anthropic_prompt.txt', 'r') as file:
-                prompt = file.read()
-            return prompt.format(retrieval_question=self.retrieval_question, context=context)
-        elif self.model_provider == "OpenAI":
-            # Generate the prompt for the Anthropic model
-            # Replace the following line with the appropriate prompt structure
-            return [
-                {
-                    "role": "system",
-                    "content": "You are a helpful AI bot that answers questions for a user. Keep your response short and direct"
-                },
-                {
-                    "role": "user",
-                    "content": context
-                },
-                {
-                    "role": "user",
-                    "content": f"{self.retrieval_question} Don't give information outside the document or repeat your findings"
-                }
-            ]
+    @abstractmethod
+    def get_prompt(self, context):
```

Comment:
Perhaps you could consider merging `get_prompt()` and `get_response_from_model()` into a single function? 

I understand the [need for precise timing](https://github.com/gkamradt/LLMTest_NeedleInAHaystack/pull/7/files#diff-05a09c80927a7bafbb8521c43d34eb8398759f066eb0e1a75a609873aaa82116R170) measurements, but I believe `get_prompt()` is so fast that it doesn't matter. 🙂

Reply:
Thank you for your suggestion regarding the merging of get_prompt() and get_response_from_model() into a single function. I appreciate the consideration given to the efficiency of the code. However, prompt engineering is a critical aspect of eliciting high-quality responses from Language Models, and the crafting of prompts can vary significantly across different models.

Maintaining a dedicated get_prompt function provides a clear and centralized location for prompt development and refinement. This separation not only ensures clarity in our codebase but also offers an organized approach for future updates and customizations specific to prompt formulation. For the time being, I believe that keeping get_prompt as an independent function is beneficial for both our current workflow and potential modifications that may arise as we continue to enhance our system's capabilities.

I hope this clarifies the rationale behind the current design choice. I'm open to further discussions on this matter as we progress and as the needs of our project evolve.